### PR TITLE
[release] Version 0.2.14

### PR DESCRIPTION
**Pull request by ereio**
### Types
- [x] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking change which improves code quality - QA thoroughly )
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (updates that would not affect or change the code involving the app itself)

#### 🔮 Features
- device rename from privacy screen @EdGeraghty 
- device rename randomizer 🎲
- action ring labeling
- copy session ID to clipboard @EdGeraghty 
- many other copy to clipboard helpers @EdGeraghty @Florian-Sabonchi 
- user sent emoji indicator
- change directory for key session backups
- enable automatic key backup schedule

#### 🐛 Fixes
- issue navigating to DM from profile preview
- plenty others!

### QA

- [ ] Signup
- [ ] Login
- [ ] Logout
- [ ] Unencrypted Chat 
- [ ] Encrypted Chat 
- [ ] Group Chats
- [ ] Profile Changes
- [ ] Theming Changes
- [ ] Force Kill and Restart

### Final Checklist
 
- [ ] All associated issues have been linked to PR
- [ ] All necessary changes made to the documentation
- [ ] Parties interested in these changes have been notified
- [ ] Linter has been run on all code in the PR

### CODE REVIEW

**Comment by FillingTheVoidOnYT** [comment]
A much needed build

File: .drone.yml [hunk 1]
```diff
@@ -131,19 +73,19 @@ steps:
       - cp -aL /usr/lib/x86_64-linux-gnu/libjsoncpp.so* build/linux/x64/release/bundle/lib/
       - cp -aL /usr/lib/x86_64-linux-gnu/libsqlite3.so build/linux/x64/release/bundle/lib/
       - cp -aL /usr/lib/x86_64-linux-gnu/libsqlcipher.so build/linux/x64/release/bundle/lib/
-      - tar -cvzf syphon.nightly.linux.x64.tar.gz -C build/linux/x64/release/bundle .
+      - tar -cvzf syphon.$(echo $DRONE_COMMIT_SHA | cut -c1-7).linux.x64.tar.gz -C build/linux/x64/release/bundle .
+      - shasum -a 256 ./syphon.$(echo $DRONE_COMMIT_SHA | cut -c1-7).linux.x64.tar.gz
 
   - name: release linux
-    image: golang:1.15
+    image: golang:1.17
```
Comment (0x1a8510f2) [inline .drone.yml:80]: Might be worth updating these to Go 1.19

**Review by dnisbetjones — COMMENTED** [review]
Thank you for pushing this through. Lots of changes!

File: .drone.yml [hunk 2]
```diff
@@ -159,7 +101,7 @@ steps:
 #    - push
 #steps:
 #  - name: build linux
-#    image: cirrusci/flutter:2.10.3
+#    image: cirrusci/flutter:3.0.1
```
Comment (0x1a8510f2) [inline .drone.yml:104]: Latest version is 3.3.3 but I'm not sure if it breaks compatibility. Might be worth upgrading though if it's a simple version bump.

File: lib/global/libs/matrix/events/requests.dart [hunk 3]
```diff
@@ -396,26 +396,29 @@ abstract class Events {
     return await json.decode(response.body);
   }
 
-  /// Send Hidden Read Receipt - unstable feature
+  /// Send Hidden Read Receipt
   ///
-  /// https://github.com/matrix-org/matrix-spec-proposals/blob/travis/msc/hidden-read-receipts/proposals/2285-hidden-read-receipts.md
-  static Future<dynamic> sendReadMarkerHidden({
+  /// https://github.com/matrix-org/matrix-spec-proposals/pull/2285
```
Comment (0x1a8510f2) [inline lib/global/libs/matrix/events/requests.dart:401]: I think this is in the spec now as of 1.4

**Comment by FillingTheVoidOnYT** [comment]
we all appreciated the hard work!

**Review by 0x1a8510f2 — APPROVED** [review]
LGTM!

I've mainly looked for things that could be outdated, since most of the changes were separately reviewed already, and I've made some comments about those, but they're mostly just small suggestions.

I am currently on mobile though so I may have missed some things.

(Lol, I didn't notice you merged already :P)

Here's to 0.2.14 🎉🥂!

File: lib/global/libs/matrix/events/requests.dart (hunk 3 above)
Comment (EdGeraghty) [reply lib/global/libs/matrix/events/requests.dart:401]: Good catch. Only actually affects the comment, though :)

File: .drone.yml (hunk 2 above)
Comment (ereio) [reply .drone.yml:104]: Intentionally left it at version 3.0.5. Going to do more rigorous testing after merging in the outstanding PRs. Been burned way too many times with bumping up minor versions of Flutter and things seemingly being ok.

File: .drone.yml (hunk 1 above)
Comment (ereio) [reply .drone.yml:80]: I wanted to upgrade this, but there's been a breaking change where the Synchronization functionality of Gitea now wipes out existing releases and make it looks exactly like the fork origin. That actually ruins the release flow we have currently. 

Looking at downgrading Gitea for now until I can figure out how to make it work with the new functionality and / or opening a issue with Gitea over it

**Pull request merged**